angularSTEP/src/app/upload-recipe/upload-recipe.component.html

```diff
@@ -1,84 +1,125 @@
-</form>
-
-
-
+<div class="stepper-container">
+<mat-horizontal-stepper linear>
+  <mat-step [stepControl]="basicsFormGroup">
+    <form [formGroup]="basicsFormGroup" (ngSubmit)="onSubmit()" class="form-formatting">
+      <ng-template matStepLabel>Recipe Basics</ng-template>
+        <mat-form-field>
+          <mat-label>Recipe Name</mat-label>
+          <input matInput placeholder="Ex. Mom's tamales" formControlName="name" required>
+          <mat-error>This is required.</mat-error>
+        </mat-form-field>
+        <h5 class="other-error"*ngIf="basicsFormGroup.controls.name.value.trim() === '' && basicsFormGroup.controls.name.touched && basicsFormGroup.controls.name.valid">Please enter a valid name.</h5>
+
+        <br><br>
+
+        <mat-form-field>
+          <mat-label>Recipe Difficulty</mat-label>
+          <mat-select formControlName="difficulty" required>
+            <mat-option value="Beginner">Beginner</mat-option>
+            <mat-option value="Intermediate">Intermediate</mat-option>
+            <mat-option value="Advanced">Advanced</mat-option>
+            <mat-option value="Professional">Professional</mat-option>
+          </mat-select>
+          <mat-error>This is required.</mat-error>
+        </mat-form-field>
+
+        <br><br>
+
+        <mat-form-field>
+          <mat-label>Short Description</mat-label>
+          <textarea matInput #description placeholder="Ex. Best Chicken tamales ever. NOT for vegetarians!" formControlName="description" required></textarea>
+          <mat-hint *ngIf="!(basicsFormGroup.controls.description.value.trim() === '' && basicsFormGroup.controls.description.touched && basicsFormGroup.controls.description.valid)">{{description.value?.length}}/200 Characters</mat-hint>
+          <mat-error>This is required.</mat-error>
+        </mat-form-field>
+        <h5 class="other-error" *ngIf="basicsFormGroup.controls.description.value.trim() === '' && basicsFormGroup.controls.description.touched && basicsFormGroup.controls.description.valid">Please enter a valid description.</h5>
+        <div id="solo-next-button">
+          <button mat-raised-button color="primary" matStepperNext [disabled]="!basicsFormGroup.valid || basicsFormGroup.controls.name.value.trim() === '' || basicsFormGroup.controls.description.value.trim() === ''" type="button">Next</button>
+        </div>
+    </form>
+  </mat-step>
+
+  <mat-step [stepControl]="ingredientsFormGroup">
+    <form [formGroup]="ingredientsFormGroup" (ngSubmit)="onSubmit()" class="form-array-formatting">
+      <ng-template matStepLabel required>Ingredients</ng-template>
+      <div formArrayName="ingredientsArray">
+        <div *ngFor="let ingred of ingredientsArray.controls; let i=index" class="array-item">
+          <mat-label>Ingredient:</mat-label>
+          <div class="input-error">
+            <input matInput type="text" [formControlName]="i">
```

Comment:
I think for the ingredients and the tools forms you might want to turn off autocomplete. This is really easy to change: adding `autocomplete="off"` to the input field, but I think it would add to userability.

Reply:
I actually might just turn off autocomplete for every input field?